# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: 6e9a89ead3a5
# Full Hash: 6e9a89ead3a52864c48a4e5481e9a35e4aad5e98
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - WebXR compositing fixes. r=imanol
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78117
# ==============================================================================

diff -r dd00e2da3a0f -r 6e9a89ead3a5 dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Wed Jun 10 22:21:02 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Wed Jun 10 22:21:04 2020 +0000
@@ -974,7 +974,6 @@
   layers::LayersBackend GetCompositorBackendType() const;
 
   bool mCapturedFrameInvalidated = false;
-  UniquePtr<layers::SurfaceDescriptor> mFrontBufferDesc;
 
   // -------------------------------------------------------------------------
   // WebGLRenderingContext Basic Properties and Methods
@@ -995,8 +994,9 @@
   }
   void GetContextAttributes(dom::Nullable<dom::WebGLContextAttributes>& retval);
 
-  void Present();
-  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(layers::TextureType);
+  void Present(WebGLFramebufferJS*, layers::TextureType);
+  Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebufferJS*,
+                                                  layers::TextureType);
   RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot() override;
 
  private: